internal/pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -157,6 +154,47 @@ func (store *updaterStore) AddDeploymentSpec(yamlDoc []byte) error {
 	return nil
 }
 
+// setAndCheckOLMNamespaces sets and checks olmTNString in dep by the
+// following logic:
+//
+// OLM places the set of target namespaces for the operator in
+// "metadata.annotations['olm.targetNamespaces']". This value should be
+// referenced (in order of preference):
+//	- In the Deployment's pod spec WATCH_NAMESPACE env variable.
+//	- In some other Deployment pod spec field.
+func setAndCheckOLMNamespaces(dep *appsv1.Deployment) bool {
+	foundWN := false
+	for _, c := range dep.Spec.Template.Spec.Containers {
```

Comment:
A loop inside of another loop usually is O(N^2) which is not good. See https://en.wikipedia.org/wiki/Big_O_notation.

Wdyt over the following suggestion?

```go
	var envContainers []string
	_ ,container  := range dep.Spec.Template.Spec.Containers {
		envContainers = append(envContainers, container.Env)
	}
	_ ,envC  := range envContainers {
		if envC.Name == k8sutil.WatchNamespaceEnvVar {
			return true
		}
	}
```

Reply:
The number of containers and environment variables is fairly small (`n < 10` for both in _most_ cases). Big O notation talks about what happens when `n` approaches some arbitrarily large `N`, which we are in no danger of approaching here; theoretically that cannot be true of course, but practically it is. As for your suggestion, creating another slice of `corev1.Env` adds memory write access overhead that is not necessary in the current implementation.

As for `foundWN`, we need to make sure _all_ instances of `WATCH_NAMESPACE` are updated and not just the first one.

Hope this explanation helps clarify why I chose this method of setting `WATCH_NAMESPACE`.